packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_impl.dart

```diff
@@ -216,15 +216,15 @@ class AmplifyStorageS3Dart extends StoragePluginInterface
       defaultPluginOptions: const S3DownloadDataPluginOptions(),
     );
 
-    final s3Options = StorageDownloadDataOptions(
+    options = StorageDownloadDataOptions(
```

Comment:
Q: why to remove final?

Reply:
it causes errors like `"Local variable 'options' can't be referenced before it is declared."` for using `options?.bucket` in the constructor, and another complaint of `The final variable 'options' can't be read because it's potentially unassigned at this point.`

Reply:
it was final before since it was called s3Options, different variable name than the one that is passed into the function as a parameter as well

Reply:
I see the issue here, it is overriding the function input parameter and it is wrong to do so. Why to need to make this change? I would suggest to reveret it.

Reply:
I did it to maintain consistency with the other functions options being called just `options` instead of `s3options.` I will revert it, I need to rebase with the most recent push to multi-bucket anyways